packages/backend-defaults/src/entrypoints/cache/providers/infinispan/InfinispanOptionsMapper.ts

```diff
@@ -0,0 +1,225 @@
+          port: serverConf.getNumber('port'),
+        } as InfinispanServerConfig;
+      } else {
+        logger?.error(
+          `Infinispan 'servers' configuration at ${storeConfigPath} must be an object or an array.`,
+        );
+        throw new Error(
+          `Infinispan 'servers' configuration at ${storeConfigPath} is invalid.`,
+        );
+      }
+    } else {
+      logger?.error(
+        `Infinispan configuration at ${storeConfigPath} is missing the 'servers' definition.`,
+      );
+      throw new Error(
+        `Infinispan configuration at ${storeConfigPath} must define 'servers'.`,
+      );
+    }
+
+    // The parsed options block to send to the Infinispan client
+    // This will be used to configure the Infinispan client behavior.
+    const behaviorOptions: Partial<InfinispanClientBehaviorOptions> = {};
+
+    if (infinispanConfig.has('clusters')) {
+      const clustersConfig = infinispanConfig.getConfigArray('clusters');
+      const clusterOptions: InfinispanClusterConfig[] = [];
+      clustersConfig?.forEach(clusterConf => {
+        const name = clusterConf.getOptionalString('name');
+        const cluster: InfinispanClusterConfig = {
+          ...(name && { name }),
+          servers: clusterConf.getConfigArray('servers').map(serverConf => ({
+            host: serverConf.getString('host'),
+            port: serverConf.getNumber('port'),
+          })),
+        };
+        clusterOptions.push(cluster);
+        behaviorOptions.clusters = clusterOptions;
+      });
+    }
+
+    // Parse Default Options Start...
+    if (infinispanConfig.has('version')) {
+      const clientVersion = infinispanConfig.getString('version');
+      if (
+        clientVersion === '2.9' ||
+        clientVersion === '2.5' ||
+        clientVersion === '2.2'
+      ) {
+        behaviorOptions.version = clientVersion;
+      } else if (clientVersion !== null && clientVersion !== undefined) {
+        logger?.warn(
+          `Invalid Infinispan client version "${clientVersion}" in config at ${storeConfigPath}.version. Must be "2.9", "2.5", or "2.2". It will be ignored, and the client may use a default or fail.`,
+        );
+      }
+    }
+
+    if (infinispanConfig.has('cacheName')) {
+      behaviorOptions.cacheName = infinispanConfig.getString('cacheName');
+    }
+
+    const mediaType = infinispanConfig.getOptionalString('mediaType');
+    if (mediaType === 'text/plain' || mediaType === 'application/json') {
+      behaviorOptions.mediaType = mediaType;
+    } else if (mediaType !== null && mediaType !== undefined) {
+      logger?.warn(
+        `Invalid Infinispan mediaType "${mediaType}" in config at ${storeConfigPath}.mediaType. Must be "text/plain" | "application/json". It will be ignored, and the client may use a default or fail.`,
+      );
+    }
+
+    if (infinispanConfig.has('maxRetries')) {
+      behaviorOptions.maxRetries = infinispanConfig.getNumber('maxRetries');
```

Comment:
since a key + undefined is normally pretty much indistinguishable from a missing key, you can often just do `behaviorOptions.maxRetries = infinispanConfig.getOptionalNumber('maxRetries');` without the `has` check.